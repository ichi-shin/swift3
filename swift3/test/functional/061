#!/bin/bash
# Copyright(c)2014 NTT corp. All Rights Reserved.
# DB1-1-2-4_15

. ./common

S3USER=tester

_s3_put /bucket -H "x-amz-grant-write-acp: id='${TENANT}:${TESTER_USER},id='${TENANT}:${TESTER2_USER}'"
_s3_get /bucket?acl -D - | _filter_curl xml

S3USER=tester2
_s3_get /bucket?acl -D - | _filter_curl xml

S3USER=tester

_s3_put /bucket?acl -H "x-amz-grant-full-control: id='${TENANT}:${TESTER_USER},id='${TENANT}:${TESTER2_USER}'"
_s3_get /bucket?acl -D - | _filter_curl xml

S3USER=tester2

_s3_get /bucket?acl -D - | _filter_curl xml
