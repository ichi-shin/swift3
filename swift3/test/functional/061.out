QA output created by 061
> s3curl --id tester -- -X PUT -H x-amz-grant-write-acp: id='TESTER,id='TESTER2' http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X GET -D - http://SWIFT_HOST/bucket?acl... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Owner>
    <ID>TESTER</ID>
    <DisplayName>TESTER</DisplayName>
  </Owner>
  <AccessControlList>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER</ID>
        <DisplayName>TESTER</DisplayName>
      </Grantee>
      <Permission>WRITE_ACP</Permission>
    </Grant>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER2</ID>
        <DisplayName>TESTER2</DisplayName>
      </Grantee>
      <Permission>WRITE_ACP</Permission>
    </Grant>
  </AccessControlList>
</AccessControlPolicy>
> s3curl --id tester2 -- -X GET -D - http://SWIFT_HOST/bucket?acl... 403
HTTP/1.1 403 Forbidden
Content-Type: text/xml
Date: DATE
Transfer-Encoding: chunked
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<Error>
  <Code>AccessDenied</Code>
  <Message>Access Denied.</Message>
  <RequestId>TXID</RequestId>
</Error>
> s3curl --id tester -- -X PUT -H x-amz-grant-full-control: id='TESTER,id='TESTER2' http://SWIFT_HOST/bucket?acl... 200
> s3curl --id tester -- -X GET -D - http://SWIFT_HOST/bucket?acl... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Owner>
    <ID>TESTER</ID>
    <DisplayName>TESTER</DisplayName>
  </Owner>
  <AccessControlList>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER</ID>
        <DisplayName>TESTER</DisplayName>
      </Grantee>
      <Permission>FULL_CONTROL</Permission>
    </Grant>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER2</ID>
        <DisplayName>TESTER2</DisplayName>
      </Grantee>
      <Permission>FULL_CONTROL</Permission>
    </Grant>
  </AccessControlList>
</AccessControlPolicy>
> s3curl --id tester2 -- -X GET -D - http://SWIFT_HOST/bucket?acl... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Owner>
    <ID>TESTER</ID>
    <DisplayName>TESTER</DisplayName>
  </Owner>
  <AccessControlList>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER</ID>
        <DisplayName>TESTER</DisplayName>
      </Grantee>
      <Permission>FULL_CONTROL</Permission>
    </Grant>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER2</ID>
        <DisplayName>TESTER2</DisplayName>
      </Grantee>
      <Permission>FULL_CONTROL</Permission>
    </Grant>
  </AccessControlList>
</AccessControlPolicy>
